code/src/LeafletProvider/DrawingTools/DrawingTools.ts

```diff
@@ -0,0 +1,230 @@
+        public polygon: boolean | any;
+        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
+        public polyline: boolean | any;
+        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
+        public rectangle: boolean | any;
+
+        constructor() {
+            this.circleMarker = false; // this tool isn't provided by our experience
+            this.circle = false;
+            this.marker = false;
+            this.polygon = false;
+            this.polyline = false;
+            this.rectangle = false;
+        }
+    }
+
+    export class DrawingTools extends OSFramework.DrawingTools
+        .AbstractDrawingTools<
+        // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any
+        any,
+        OSFramework.Configuration.IConfigurationDrawingTools
+    > {
+        // FeatureGroup <any>: as required by Leaflet
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        private _toolsGroup: L.FeatureGroup<any>;
+        protected _provider: google.maps.drawing.DrawingManager;
+
+        constructor(
+            map: OSFramework.OSMap.IMap,
+            drawingToolsId: string,
+            configs: Configuration.DrawingTools.DrawingToolsConfig
+        ) {
+            super(
+                map,
+                drawingToolsId,
+                new Configuration.DrawingTools.DrawingToolsConfig(configs)
+            );
+
+            this._toolsGroup = new L.FeatureGroup();
+        }
+
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        private _addCompletedEventHandler(event: any): void {
+            const toolType = event.layerType;
+
+            this.tools
+                .filter((element) => element.type === toolType)
+                .forEach((element) => element.addCompletedEvent(event));
+        }
+
+        private _getDrawingToolsPosition(
+            position: string
+        ): Constants.DrawingTools.Positions {
+            if (Constants.DrawingTools.Positions[position] !== undefined) {
+                return Constants.DrawingTools.Positions[position];
+            } else {
+                OSFramework.Helper.ThrowError(
+                    this.map,
+                    OSFramework.Enum.ErrorCodes.CFG_InvalidDrawingToolsPosition,
+                    `${position}`
+                );
+                return;
+            }
+        }
+
+        //TODO: create the return structure
+        private _getTools(): ToolsList {
+            const _tools = new ToolsList();
+            this.tools.forEach((tool) => {
+                switch (tool.type) {
+                    case OSFramework.Enum.DrawingToolsTypes.Circle:
+                        _tools.circle = tool.options || {};
```

Comment:
maybe options it self should return an `{}` in case there's nothing to be returned. Alternatively create a `const` before the `switch`...

Reply:
tool.options can return the following values:
- false (tool disabled)
- object (tool enabled) if the object is {}, it is enabled but with the defaults

Reply:
if it returns false then it's the same as returning `{}` right away...